File: src/pyglue/tests/ColorSpaceTest.py

import unittest, os, sys
sys.path.append(os.path.join(sys.argv[1], "src", "pyglue"))
import PyOpenColorIO as OCIO

class ColorSpaceTest(unittest.TestCase):
    
    def test_interface(self):
        cs = OCIO.ColorSpace()
        cs.setName("mynewcolspace")
        self.assertEqual("mynewcolspace", cs.getName())
        cs.setFamily("fam1")
        self.assertEqual("fam1", cs.getFamily())
        cs.setEqualityGroup("match1")
        self.assertEqual("match1", cs.getEqualityGroup())
        cs.setDescription("this is a test")
        self.assertEqual("this is a test", cs.getDescription())
        cs.setBitDepth(OCIO.Constants.BIT_DEPTH_F16)
        self.assertEqual(OCIO.Constants.BIT_DEPTH_F16, cs.getBitDepth())
        cs.setIsData(False)
        self.assertEqual(False, cs.isData())
        cs.setAllocation(OCIO.Constants.ALLOCATION_LG2)
        self.assertEqual(OCIO.Constants.ALLOCATION_LG2, cs.getAllocation())
        cs.setAllocationVars([0.1, 0.2, 0.3])
        self.assertEqual(3, len(cs.getAllocationVars()))
        lt = OCIO.LogTransform()
        lt.setBase(10.0)
        cs.setTransform(lt, OCIO.Constants.COLORSPACE_DIR_TO_REFERENCE)
        ott = cs.getTransform(OCIO.Constants.COLORSPACE_DIR_TO_REFERENCE)
        self.assertEquals(10.0, ott.getBase())

File: src/java/tests/org/OpenColorIO/ColorSpaceTest.java

import junit.framework.TestCase;
import org.OpenColorIO.*;

public class ColorSpaceTest extends TestCase {
    
    protected void setUp() {
    }
    
    protected void tearDown() {
    }
    
    public void test_interface() {
        ColorSpace cs = new ColorSpace().Create();
        cs.setName("mynewcolspace");
        assertEquals("mynewcolspace", cs.getName());
        cs.setFamily("fam1");
        assertEquals("fam1", cs.getFamily());
        cs.setEqualityGroup("match1");
        assertEquals("match1", cs.getEqualityGroup());
        cs.setDescription("this is a test");
        assertEquals("this is a test", cs.getDescription());
        cs.setBitDepth(BitDepth.BIT_DEPTH_F16);
        assertEquals(BitDepth.BIT_DEPTH_F16, cs.getBitDepth());
        cs.setIsData(false);
        assertEquals(false, cs.isData());
        cs.setAllocation(Allocation.ALLOCATION_LG2);
        assertEquals(Allocation.ALLOCATION_LG2, cs.getAllocation());
        float test[] = new float[]{0.1f, 0.2f, 0.3f};
        cs.setAllocationVars(3, test);
        assertEquals(3, cs.getAllocationNumVars());
        float out[] = new float[3];
        cs.getAllocationVars(out);
        LogTransform lt = new LogTransform().Create();
        lt.setBase(10.f);
        cs.setTransform(lt, ColorSpaceDirection.COLORSPACE_DIR_TO_REFERENCE);
        LogTransform ott = (LogTransform)cs.getTransform(ColorSpaceDirection.COLORSPACE_DIR_TO_REFERENCE);
        assertEquals(10.f, ott.getBase());
    }
    
}
